crowbar_framework/app/models/tempest_service.rb

```diff
@@ -94,6 +94,11 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
 
     role.save
 
+    # Allocate a public IP, tempest needs it
+    tempest_elements, tempest_node, tempest_ha_enabled = role_expand_elements(role, "tempest")
```

Comment:
Also, you need to check that `tempest_node` is not empty (can happen when deactivating).